lib/fastifySession.js

```diff
@@ -115,7 +116,10 @@ function onRequest (options) {
       done()
       return
     }
-    const sessionId = request.cookies[options.cookieName]
+    let sessionId = request.cookies[options.cookieName]
+    if (sessionId && cookiePrefix && sessionId.slice(0, cookiePrefix.length) === cookiePrefix) {
```

Comment:
Can we have a benchmark for cookiePrefix? I once wrote a jit startswith https://github.com/fastify/help/issues/711 maybe using this instead of slicing new strings is improving the performance.

Reply:
First, running on main on my local computer:

```
% yarn benchmark
yarn run v1.22.10
$ node benchmark/bench.js
╔══════════════╤═════════╤═══════════════╤═══════════╤═════════════════════════╗
║ Slower tests │ Samples │        Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ file         │      25 │  32.68 op/sec │ ±  7.20 % │                         ║
║ redis        │      25 │ 139.43 op/sec │ ± 18.75 % │ + 326.64 %              ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ Fastest test │ Samples │        Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ memory       │      25 │ 760.16 op/sec │ ± 38.32 % │ + 2225.96 %             ║
╚══════════════╧═════════╧═══════════════╧═══════════╧═════════════════════════╝

✨  Done in 27.04s.
```

Below are the results from running on this branch, without benchmark using cookeiPrefix.

One of the reasons I had `cookiePrefix` first in the `if` is for performance. This way, `sessionId.slice(0, cookiePrefix.length) === cookiePrefix` doesn't get run unless cookiePrefix is set.
```
% yarn benchmark                
yarn run v1.22.10
$ node benchmark/bench.js
╔══════════════╤═════════╤═══════════════╤═══════════╤═════════════════════════╗
║ Slower tests │ Samples │        Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ file         │      25 │  32.72 op/sec │ ±  7.20 % │                         ║
║ redis        │      25 │ 139.51 op/sec │ ± 18.10 % │ + 326.39 %              ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ Fastest test │ Samples │        Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ memory       │      25 │ 894.27 op/sec │ ± 31.78 % │ + 2633.27 %             ║
╚══════════════╧═════════╧═══════════════╧═══════════╧═════════════════════════╝

✨  Done in 27.01s.
```

Finally, running benchmark with `s:` in cookiePrefix:

```
yarn run v1.22.10
$ node benchmark/bench.js
╔══════════════╤═════════╤═══════════════╤═══════════╤═════════════════════════╗
║ Slower tests │ Samples │        Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ file         │      25 │  33.57 op/sec │ ±  8.04 % │                         ║
║ redis        │      25 │ 180.14 op/sec │ ± 16.00 % │ + 436.69 %              ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ Fastest test │ Samples │        Result │ Tolerance │ Difference with slowest ║
╟──────────────┼─────────┼───────────────┼───────────┼─────────────────────────╢
║ memory       │      25 │ 885.81 op/sec │ ± 30.11 % │ + 2539.07 %             ║
╚══════════════╧═════════╧═══════════════╧═══════════╧═════════════════════════╝

✨  Done in 26.57s.
```

Again, no real difference.